miio/integrations/vacuum/dreame/dreamevacuum_miot.py

```diff
@@ -233,51 +418,120 @@ def status(self) -> DreameVacuumStatus:
-        return self.send_action(17, 1)
+        return self.call_action("locate")
 
     @command()
     def reset_mainbrush_life(self) -> None:
         """Reset main brush life."""
-        return self.send_action(26, 1)
+        return self.call_action("reset_mainbrush_life")
 
     @command()
     def reset_filter_life(self) -> None:
         """Reset filter life."""
-        return self.send_action(27, 1)
+        return self.call_action("reset_filter_life")
 
     @command()
     def reset_sidebrush_life(self) -> None:
         """Reset side brush life."""
-        return self.send_action(28, 1)
+        return self.call_action("reset_sidebrush_life")
+
+    @command()
+    def play_sound(self) -> None:
+        """Play sound."""
+        return self.call_action("play_sound")
+
+    @command()
+    def fan_speed_presets(self) -> Dict[str, int]:
+        """Return dictionary containing supported fan speeds."""
+
+        def _enum_as_dict(cls):
+            return {x.name: x.value for x in list(cls)}
+
+        fanspeeds_enum = self.get_fanspeeds()
+        if not fanspeeds_enum:
+            return {}
+        return _enum_as_dict(fanspeeds_enum)
+
+    @command(
+        click.argument("distance", default=30, type=int),
+    )
+    def forward(self, distance: int) -> None:
+        """Move forward."""
+        if distance < self.MANUAL_DISTANCE_MIN or distance > self.MANUAL_DISTANCE_MAX:
+            raise DeviceException(
+                "Given distance is invalid, should be [%s, %s], was: %s"
+                % (self.MANUAL_DISTANCE_MIN, self.MANUAL_DISTANCE_MAX, distance)
+            )
+        self.call_action(
+            "move",
+            [
+                {
+                    "piid": 1,
+                    "value": "0",
+                },
+                {
+                    "piid": 2,
+                    "value": f"{distance}",
+                },
+            ],
+        )
+
+    @command(
+        click.argument("rotatation", default=90, type=int),
+    )
+    def rotate(self, rotatation: int) -> None:
+        """Rotate vacuum."""
+        if (
+            rotatation < self.MANUAL_ROTATION_MIN
+            or rotatation > self.MANUAL_ROTATION_MAX
+        ):
+            raise DeviceException(
+                "Given rotation is invalid, should be [%s, %s], was %s"
+                % (self.MANUAL_ROTATION_MIN, self.MANUAL_ROTATION_MAX, rotatation)
+            )
+        self.call_action(
+            "move",
+            [
+                {
+                    "piid": 1,
+                    "value": f"{rotatation}",
+                },
+                {
+                    "piid": 2,
+                    "value": "0",
+                },
+            ],
+        )
+
+
+class DreameVacuum(DreameVacuumBase):
+    """Interface for Vacuum 1C STYTJ01ZHM (dreame.vacuum.mc1808)"""
+
+
+Dreame1CVacuum = DreameVacuum
+DreameF9Vacuum = DreameVacuum
+DreameVacuumMiot = DreameVacuum
```

Comment:
Please just rename `DreamVacuumBase` to `DreameVacuum`. You can leave `DreameVacuumMiot` or create a class that just inherits from the `DreameVacuum` and is decorated with `@deprecated()` just to let any potential users that it's been renamed, but it's not worth adding new aliases.

In practice, this means that using `miiocli dreamevacuum` should work directly on all supported vacuums.